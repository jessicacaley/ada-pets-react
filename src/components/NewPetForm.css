.new-pet-form--label {
  margin-right: 30px;
  margin-left: 15px;
  width: 50px;
  font-weight: bold;
}

.new-pet-form--submit {
  margin-left: 15px;
}

.new-pet-form {
  margin: 15px;
}

.new-pet-form--about {
  width: 400px;
  margin: 10px;
  height: 200px;
}

.invalid-image-form::placeholder {
  color: #ffb2b2;
}