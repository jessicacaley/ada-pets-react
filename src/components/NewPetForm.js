import React, { Component } from 'react';
import PropTypes from 'prop-types';

import 'bootstrap/dist/css/bootstrap.min.css';
import './NewPetForm.css'

class NewPetForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '', // required
      species: '', // required
      location: '',
      images: [], // required
      about: '',
    };
  }
  
  onNameChange = (event) => {
    console.log(`Name Field updated ${event.target.value}`);
    this.setState({
      name: event.target.value,
    });
  }

  onSpeciesChange = (event) => {
    console.log(`Species Field updated ${event.target.value}`);
    this.setState({
      species: event.target.value,
    });
  }

  onLocationChange = (event) => {
    console.log(`Location Field updated ${event.target.value}`);
    this.setState({
      location: event.target.value,
    });
  }

  onImageChange = (event) => {
    console.log(`Image Field updated ${event.target.value}`);
    this.setState({
      images: event.target.value,
    });
  }

  onAboutChange = (event) => {
    console.log(`About Field updated ${event.target.value}`);
    this.setState({
      about: event.target.value,
    });
  }

  formValid = () => {
    const imagesValid = /(http(s?):)([/|.|\w|\s|-])*\.(?:jpg|gif|png|jpeg)/.test(this.state.images);
    // this regex is based off of https://regexr.com/3g1v7

    if(imagesValid) {
      return true
    } else {
      return false
    }
  }

  imagesPlaceholderText = 'image link'
  invalidImage = ''

  onFormSubmit = (event) => {
    event.preventDefault();

    if(this.formValid()) {
      const newPet = {
        name: this.state.name,
        species: this.state.species,
        location: this.state.location,
        images: Array(this.state.images),
        about: this.state.about,
      };

      console.log(this.state)
    
      this.setState({
        name: '', // required
        species: '', // required
        location: '',
        images: '', // required
        about: '',
      });
    
      this.props.addPetCallback(newPet);
    } else {
      this.imagesPlaceholderText = 'must be valid url!';
      this.invalidImage = 'invalid-image-form'
      this.setState({
        images: '',
      })
    }
  }

  render() {
    return (
      <form  className="new-pet-form" onSubmit={this.onFormSubmit}>
        <h3>Add a Pet</h3>
        <div>
          <label htmlFor="name" className="new-pet-form--label">Name</label>
          <input 
            onChange={this.onNameChange}
            value={this.state.name}
            name="name"
            type="text"
            placeholder="name" 
            required
             />
        </div>
        <div>
          <label htmlFor="species" className="new-pet-form--label">Species</label>
          <input 
            onChange={this.onSpeciesChange}
            value={this.state.species}
            name="species" 
            type="text" 
            placeholder="species"
            required />
        </div>
        <div>
          <label htmlFor="location" className="new-pet-form--label">Location</label>
          <input 
            onChange={this.onLocationChange}
            value={this.state.location}
            name="location" 
            type="text" 
            placeholder="location"/>
        </div>
        <div>
          <label htmlFor="images" className="new-pet-form--label">Image</label>
          <input 
            className={this.invalidImage}
            onChange={this.onImageChange}
            value={this.state.images}
            name="images" 
            type="text" 
            placeholder={this.imagesPlaceholderText}
            required />
        </div>
        <div>
          <label htmlFor="about" className="new-pet-form--label">About</label>
        </div>
        <div>
          <textarea 
            onChange={this.onAboutChange}
            value={this.state.about}
            name="about" 
            form="new-pet-form" 
            className="new-pet-form--about"></textarea>
        </div>
       
        <input className="btn btn-success new-pet-form--submit" type="submit" name="submit" value="Add a Pet" />
      </form>
    );
  }


}

NewPetForm.propTypes = {
  addPetCallback: PropTypes.func.isRequired,
};

export default NewPetForm;