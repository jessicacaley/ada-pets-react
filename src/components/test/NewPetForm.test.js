import React from 'react';
import NewPetForm from '../NewPetForm';
import {shallow } from 'enzyme';

describe('NewPetForm', () => {
  test('that it matches an existing snapshot', () => {
    const wrapper = shallow ( <NewPetForm addPetCallback={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });
});