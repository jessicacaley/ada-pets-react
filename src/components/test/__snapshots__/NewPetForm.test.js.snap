// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPetForm that it matches an existing snapshot 1`] = `
<form
  className="new-pet-form"
  onSubmit={[Function]}
>
  <h3>
    Add a Pet!
  </h3>
  <div>
    <label
      className="new-pet-form--label"
      htmlFor="name"
    >
      Name
    </label>
    <input
      name="name"
      onChange={[Function]}
      placeholder="name"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      className="new-pet-form--label"
      htmlFor="species"
    >
      Species
    </label>
    <input
      name="species"
      onChange={[Function]}
      placeholder="species"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      className="new-pet-form--label"
      htmlFor="location"
    >
      Location
    </label>
    <input
      name="location"
      onChange={[Function]}
      placeholder="location"
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      className="new-pet-form--label"
      htmlFor="images"
    >
      Image
    </label>
    <input
      className=""
      name="images"
      onChange={[Function]}
      placeholder="image link"
      required={true}
      type="text"
      value={Array []}
    />
  </div>
  <div>
    <label
      className="new-pet-form--label"
      htmlFor="about"
    >
      About
    </label>
  </div>
  <div>
    <textarea
      className="new-pet-form--about"
      form="new-pet-form"
      name="about"
      onChange={[Function]}
      value=""
    />
  </div>
  <input
    className="btn btn-success new-pet-form--submit"
    name="submit"
    type="submit"
    value="Add a Pet"
  />
</form>
`;
